File: tradingagents_web/schemas/fx.py
"""Pydantic schemas for the FX (foreign exchange) API."""
from datetime import date, datetime

from pydantic import BaseModel


class FxRate(BaseModel):
    """USD/KRW spot rate snapshot.

    rate is None when the upstream lookup failed and no prior cache existed.
    """

    pair: str  # "USDKRW"
    rate: float | None
    as_of: date | None
    fetched_at: datetime

File: web/lib/fx.ts
import { api } from "./api";

export interface FxRate {
  pair: "USDKRW";
  rate: number | null;
  as_of: string | null;
  fetched_at: string;
}

const BASE = process.env.NEXT_PUBLIC_API_BASE ?? "";

export async function getUsdKrwRate(): Promise<FxRate> {
  return api(`${BASE}/api/fx/usd-krw`);
}
